tools/lsp/fmt/fmt.rs

```diff
@@ -742,10 +756,26 @@ fn format_array(
             }
         })
         .unwrap_or(false);
+    // len of all childrens
```

Comment:
this is the unformated len of the children.
This is really difficult to do right efficiently. We could try to get the size of each token (not node) without any whitespace and then just add some space. Also it would looke strange if the inner is formatted on several line, but not the outer.  Take for example: `[[1], [1, 2,], [1, 2, 3]]`

anyway, this is good enough heuristics for now.